runtime/fieldmask.go

```diff
@@ -48,23 +47,3 @@ type fieldMaskPathItem struct {
 	// a generic decoded json object the current item to inspect for further path extraction
 	node interface{}
 }
-
-// CamelCaseFieldMask updates the given FieldMask by converting all of its paths to CamelCase, using the same heuristic
-// that's used for naming protobuf fields in Go.
-func CamelCaseFieldMask(mask *field_mask.FieldMask) {
```

Comment:
This func being removed, and subsequently released as a patch release (v1.11.3), broke semantic versioning guarantees (our code no longer builds simply by updating from v1.11.2 to v1.11.3). Could you please avoid doing this in the future?

Reply:
Terribly sorry @RJPercival, you're right about this breaking users using a generator from version v1.11.2 and a runtime library with version v1.11.3. _However_, we do always encourage our users to ensure their generator versions and runtime library versions match up, so this shouldn't be a problem for users that can upgrade both at the same time. Sorry for the inconvenience caused.

Reply:
I don't think this can be guaranteed with Go Modules. You may import a dependency that used an older version of the generator and Go Modules will happily use a newer runtime library, so long as it's at the same major version, since that's supposed to mean it's backwards compatible.

Reply:
This can (and should) be guaranteed by go modules. Define a [tool dependency](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) for the generator, and run `go install github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway` before invoking `protoc`.

Reply:
I agree that it's not ideal though, and I apologize for the inconvenience caused.

Reply:
I don't think that guarantees what you think it guarantees. We already define a tool dependency, and have grpc-gateway listed in our go.mod file with a pre-v1.11.3 version number. However, according to [the version selection rules of Go Modules](https://github.com/golang/go/wiki/Modules#version-selection), anyone can import our module and require a newer v1 version of grpc-gateway in their go.mod file and it'll use that newer version (because it's supposed to be backwards-compatible), resulting in a build error. They then have to figure out that v1.11.3 of grpc-gateway broke them and pin to an earlier version.

Thanks for the apology. I don't mean to labour the point, but I just want to make it clear that violating semantic versioning rules **will** break your users and so we'd very much appreciate you avoiding it, thanks! :slightly_smiling_face: